File: Practicas/Practica2/Potencia2.java
package Practicas.Practica2;
import java.util.Scanner;
/**
 *
 * @author Salas Chavez Jose de Jesus
 */
public class Potencia2 {
   
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num,aux,count=0;
        System.out.print("Ingrese un numero: ");
        num = scanner.nextInt();
        aux=num;
        while(aux!=1){
            aux=aux/2;
            count++;
        }
        System.out.print("Tu numero es: "+num+"\n");
        System.out.print("La potencia mas cercana es: 2^"+count+" = "+Math.pow(2,count)+"\n");
    }
   
}

File: Practicas/Practica3/Potencia2.py
"""
Autor: Salas Chavez Jose de Jesus
"""
def main():
    num = int(input("Ingrese un número: "))
    aux = num
    count = 0
    
    while aux != 1:
        aux = aux // 2  # División entera
        count += 1
    
    print(f"Tu número es: {num}")
    print(f"La potencia más cercana es: 2^{count} = {2 ** count}")

if __name__ == "__main__":
    main()